### PR TITLE
FaF support request - updates to DfE signed in journey for MATs

### PR DESCRIPTION
**Pull request by PiperS52**
## Changes in this PR

- update content for kick-out page 
- add trusts to the DSI journey school page
- add `(MAT)` string to trusts listed on DSI journey school page

### Before

image

### After

image
image

### CODE REVIEW

**Review by kennyevil — APPROVED** [review]

File: app/forms/framework_support_form_schema.rb [hunk 1]
```diff
@@ -41,15 +41,30 @@ class FrameworkSupportFormSchema < Schema
...
-    key.failure(:missing) if key? && values[:group] && Support::EstablishmentGroup.find_by(uid: value.split(" - ").first).nil?
+    # validate individual group_uid field only in the non-dsi journey
+    key.failure(:missing) if key? && !values[:dsi] && values[:group] && !org_exists?(value.split(" - ").first, true)
   end
 
   rule(:message_body) do
     key.failure(:missing) if key? && value.blank?
   end
+
+  rule do
```
Comment (peterdavidhamilton) [inline app/forms/framework_support_form_schema.rb:57]: Can you change it so there is an `affiliation` rule with a `missing` key please.

File: app/forms/framework_support_form_schema.rb [hunk 2]
```diff
@@ -41,15 +41,30 @@ class FrameworkSupportFormSchema < Schema
...
+    # validate that either school_urn or group_uid is provided in the dsi journey
+    base.failure(:missing_org) if values[:dsi] && values[:school_urn].blank? && values[:group_uid].blank?
+  end
+
+  def org_exists?(id, group)
+    if group
+      Support::EstablishmentGroup.find_by(uid: id).present?
```
Comment (peterdavidhamilton) [inline app/forms/framework_support_form_schema.rb:64]: Same thing here about accessing supported data. What I'd like is a function that if provided an identifier will return a hash of everything we need for use in Specify if it looked the same as the DSI orgs array we would have a common interface. Add a note same as in the controller.

File: app/views/framework_requests/show.html.erb [hunk 3]
```diff
@@ -51,7 +51,7 @@
         <% end %>
         <dd class="govuk-summary-list__actions">
           <% unless @framework_request.submitted? %>
-            <%= link_to I18n.t("generic.button.change_answer"), edit_framework_request_path(@framework_request, step: 3), class: "govuk-link", id: "edit-school" unless current_user.supported_schools.one? %>
+            <%= link_to I18n.t("generic.button.change_answer"), edit_framework_request_path(@framework_request, step: 3), class: "govuk-link", id: "edit-school" unless current_user.supported_orgs.one? %>
```
Comment (peterdavidhamilton) [inline app/views/framework_requests/show.html.erb:54]: `single_org?`

File: app/views/framework_requests/steps/_step_3.html.erb [hunk 4]
```diff
@@ -18,15 +18,19 @@ form_data = @framework_support_form.to_h.merge(step: @framework_support_form.ste
...
+        <% if org.urn %>
+          <%= form.govuk_radio_button :school_urn, org.urn, label: { text: org.name } %>
+        <% else %>
+          <%= form.govuk_radio_button :group_uid, org.uid, label: { text: org.name } %>
+        <% end %>
+      <% end %>
+  <% end %>
+
+  <%= javascript_include_tag "faf_select_school" %>
 <% elsif form.object.group == true %>
```
Comment (peterdavidhamilton) [inline app/views/framework_requests/steps/_step_3.html.erb:32]: We could drop the `== true` but this is an opportunity to build the interface of the form, so we could have:

```ruby
def multiple_schools?
  instance_variable_get :@group
end
```

Be aware the `group` attribute has been duplicated in the form

**Review by peterdavidhamilton — CHANGES_REQUESTED** [review]

**Review by peterdavidhamilton — APPROVED** [review]

**Pull request merged**